lib/ufe/StagesSubject.cpp

```diff
@@ -223,5 +254,39 @@ void StagesSubject::onStageSet(const UsdMayaProxyStageSetNotice& notice)
 	afterOpen();
 }
 
+AttributeChangedNotificationGuard::AttributeChangedNotificationGuard()
+{
+    if (inAttributeChangedNotificationGuard()) {
+        TF_CODING_ERROR("Attribute changed notification guard cannot be nested.");
+    }
+
+    if (attributeChangedNotificationGuardCount == 0 &&
+        !pendingAttributeChangedNotifications.empty()) {
+        TF_CODING_ERROR("Stale pending attribute changed notifications.");
+    }
+
+    ++attributeChangedNotificationGuardCount;
+
+}
+
+AttributeChangedNotificationGuard::~AttributeChangedNotificationGuard()
+{
+    --attributeChangedNotificationGuardCount;
+
+    if (attributeChangedNotificationGuardCount < 0) {
```

Comment:
If you want, these two lines can be condensed to a single line: 
    if (--attributeChangedNotificationGuardCount < 0) {